sormas-ui/src/main/java/de/symeda/sormas/ui/utils/VaadinUiUtil.java

```diff
@@ -135,15 +138,42 @@ public static Window showPopupWindowWithWidth(Component content, String caption,
 	}
 
 	public static Window showModalPopupWindow(CommitDiscardWrapperComponent<?> content, String caption) {
-		return showModalPopupWindow(content, caption, null);
+		return showModalPopupWindow(content, caption, false);
+	}
+
+	public static Window showModalPopupWindow(CommitDiscardWrapperComponent<?> content, String caption, boolean discardOnClose) {
+		return showModalPopupWindow(content, caption, discardOnClose, null);
 	}
 
 	public static Window showModalPopupWindow(CommitDiscardWrapperComponent<?> content, String caption, Consumer<Boolean> callback) {
+		return showModalPopupWindow(content, caption, false, callback);
+	}
+
+	public static Window showModalPopupWindow(
+		CommitDiscardWrapperComponent<?> content,
+		String caption,
+		boolean discardOnClose,
+		Consumer<Boolean> callback) {
 		final Window popupWindow = VaadinUiUtil.showPopupWindow(content);
 		popupWindow.setCaption(caption);
 		content.setMargin(true);
 
-		content.addDoneListener(popupWindow::close);
+		final Registration closeListener;
+		if (discardOnClose) {
+			closeListener = popupWindow.addCloseListener((e) -> {
+				content.discard();
```

Comment:
Why is it necessary to call this?

Reply:
It is needed because of the usage of CompletableFuture that must be completed.

In case the user simply closes the dialog by clicking the `X` icon on the top left corner, there is no handler called if this listener is not added. No handler called means that the future never completes it keeps the thread running forever causing memory leaks and performance issues. 

This way the on close the discard handler is forced so the future will complete.